.github/workflows/tests.yaml

```diff
@@ -0,0 +1,34 @@
+name: Tests
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+# Restrict tests to the most recent commit.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    container: debian:bookworm
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        shell: bash
+        run: |
+          echo "deb http://deb.debian.org/debian/ bookworm-backports main" > /etc/apt/sources.list.d/bookworm-backports.list
+          echo "deb-src http://deb.debian.org/debian/ bookworm-backports main" >> /etc/apt/sources.list.d/bookworm-backports.list
+          apt-get update
+          apt-get install golang-1.22
```

Comment:
I think if we're not using our Dockerfiles here, then we should use the https://github.com/actions/setup-go actions.

Reply:
Better link: https://github.com/marketplace/actions/setup-go-environment

Reply:
This was not ready for review yet - I was just hoping to play around in CI. Maybe I should do this on a fork first so that I'm not spamming y'all :)

Reply:
Oh, you can still run CI if you make this a draft PR! Then we don't get CODEOWNERS stuff happening